recipes/googleapis/all/conanfile.py

```diff
@@ -11,18 +18,126 @@ class GoogleAPIS(ConanFile):
     homepage = "https://github.com/googleapis/googleapis"
     topics = "google", "protos", "api"
     settings = "os", "arch", "compiler", "build_type"
-    short_paths = True
+    generators = "cmake", "cmake_find_package"
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False]
+        }
+    default_options = {
+        "shared": False, 
+        "fPIC": True
+        }
+    exports = "helpers.py"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
 
     def source(self):
         get(self, **self.conan_data["sources"][str(self.version)], destination=self.source_folder, strip_root=True)
 
-    def package_id(self):
-        self.info.header_only()
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) <= "5":
+            raise ConanInvalidConfiguration("Build with GCC 5 fails")
+        if self.options.shared and not self.options["protobuf"].shared:
+            raise ConanInvalidConfiguration("If built as shared, protobuf must be shared as well. Please, use `protobuf:shared=True`")
```

Comment:
Otherwise when it is linked into the executable (`test_package`) the same proto definition arrives from two different libraries:
 * from `protobuf` (static) linked into `googleapis` (shared)
 * from `protobuf` linked directly into `test_package`.

It links successfully, but protobuf runs some checks at startup and realizes about this duplication. You can see the logs from the latest failure for `shared=True`:

```
[libprotobuf ERROR /Users/jenkins/w/prod/BuildSingleReference/.conan/data/protobuf/3.21.1/_/_/build/a68b48337784cbbfa52d25f15d209e461efe2cbf/source_subfolder/src/google/protobuf/descriptor_database.cc:642] File already exists in database: google/protobuf/wrappers.proto
```